backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/GoogleMeetService.cs

```diff
@@ -0,0 +1,149 @@
+﻿using AutoMapper;
+using EasyMeets.Core.BLL.Helpers;
+using EasyMeets.Core.BLL.Interfaces;
+using EasyMeets.Core.Common.DTO.Calendar;
+using EasyMeets.Core.Common.DTO.Credentials.GoogleMeet;
+using EasyMeets.Core.Common.DTO.GoogleMeet;
+using EasyMeets.Core.Common.Enums;
+using EasyMeets.Core.DAL.Context;
+using EasyMeets.Core.DAL.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+
+namespace EasyMeets.Core.BLL.Services;
+
+public class GoogleMeetService : BaseService, IGoogleMeetService
+{
+    private readonly IGoogleOAuthService _googleOAuthService;
+    private readonly IConfiguration _configuration;
+    private readonly IUserService _userService;
+    public GoogleMeetService(EasyMeetsCoreContext context, IMapper mapper, IGoogleOAuthService googleOAuthService, IConfiguration configuration, IUserService userService) : base(context, mapper)
+    {
+        _googleOAuthService = googleOAuthService;
+        _configuration = configuration;
+        _userService = userService;
+    }
+
+    public async Task<List<MeetCredentialsDto>> GetAvailableEmails()
+    {
+        var userId = _userService.GetCurrentUserId();
+        return await _context.Calendars
+            .Include(c => c.User)
+            .Where(c => c.User.Uid == userId)
+            .Select(c => new MeetCredentialsDto
+            {
+                UserEmail = c.ConnectedCalendar
+            }).ToListAsync();
+    }
+
+    public async Task<MeetCredentialsDto?> GetCurrentCredentials()
+    {
+        var userId = _userService.GetCurrentUserId();
+        return await _context.Credentials
+            .Include(c => c.User)
+            .Where(c => c.User.Uid == userId && c.Type == CredentialsType.Meet)
+            .Select(c => _mapper.Map<MeetCredentialsDto>(c))
+            .FirstOrDefaultAsync();
+    }
+
+    public Task DeleteCredentials(string email, long userId)
+    {
+        var credentials = _context.Credentials
+            .Where(c => c.UserId == userId &&
+                        c.UserEmail == email &&
+                        c.Type == CredentialsType.Meet);
+        _context.Credentials.RemoveRange(credentials);
+        return _context.SaveChangesAsync();
+    }
+
+    public async Task CreateDefaultCredentials(string? userId = null)
+    {
+        var user = await _context.Users
+                       .Include(u => u.Credentials)
+                       .Include(u => u.Calendars)
+                       .FirstOrDefaultAsync(u => u.Uid == (userId ?? _userService.GetCurrentUserId())) 
```

Comment:
I think you should compute user id outside of the query in order to not call GetCurrentUserId() each time

Reply:
Fixed

Reply:
spoiler: it will not call each time GetCurrentUserId (: